fe/fe-core/src/main/java/com/starrocks/alter/CompactionHandler.java

```diff
@@ -88,29 +93,28 @@ public synchronized ShowResultSet process(List<AlterClause> alterClauses, Databa
                                               OlapTable olapTable) throws UserException {
         Preconditions.checkArgument(alterClauses.size() == 1);
         AlterClause alterClause = alterClauses.get(0);
-        if (alterClause instanceof CompactionClause) {
-            CompactionClause compactionClause = (CompactionClause) alterClause;
+        if (!(alterClause instanceof CompactionClause)) {
+            Preconditions.checkState(false, alterClause.getClass());
+            return null;
+        }
+
+        CompactionClause compactionClause = (CompactionClause) alterClause;
+        if (RunMode.isSharedDataMode()) {
+            List<Partition> allPartitions = findAllPartitions(olapTable, compactionClause);
```

Comment:
should process physical partition, check logic in line 114

Reply:
Lake compaction scheduler will use physical partitions by default. Refer to [code](https://github.com/StarRocks/starrocks/blob/b49544e2d5eccba5548a301df39ac60a9ea7c5b4/fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionScheduler.java#L274)